server/src/main/java/org/elasticsearch/index/codec/vectors/diskbbq/IVFVectorsReader.java

```diff
@@ -43,20 +42,37 @@
 
 import static org.apache.lucene.codecs.lucene99.Lucene99HnswVectorsReader.SIMILARITY_FUNCTIONS;
 import static org.elasticsearch.index.codec.vectors.diskbbq.ES920DiskBBQVectorsFormat.DYNAMIC_VISIT_RATIO;
+import static org.elasticsearch.index.codec.vectors.diskbbq.ES920DiskBBQVectorsFormat.VERSION_DIRECT_IO;
 
 /**
  * Reader for IVF vectors. This reader is used to read the IVF vectors from the index.
  */
 public abstract class IVFVectorsReader extends KnnVectorsReader {
 
+    private record FlatVectorsReaderKey(String formatName, boolean useDirectIO) {
```

Comment:
Maybe put it in its own class? Wouldn't we want this same thing for other formats eventually?

Reply:
Currently, `ES93GenericFlatVectorsReader` only stores one generic format name across all fields. Is this actually incorrect, and we need to use the same per-field things in generic HNSW formats as used here?